using/self_service.md

```diff
@@ -322,3 +322,118 @@ Once allocation has been created, you can see its access information:
+2. Click Details on the left side of the screen
+3. Select SSH keys
+4. Click Add SSH key
+5. Paste there contents of a public SSH key
+
+![Adding public SSH key](../images/adding-ssh-key.png)
+
+## Python management service
+
+### Prerequisites
+
+1. Your project in ETAIS has a Private Cloud. Please refer to [Adding a VPC](#adding-a-vpc)
+2. Your account has your SSH public key. Please refer to [Adding public SSH key](#adding-a-public-ssh-key-to-a-profile)
+3. Virtual machine is create and properly configured. For details please refer to [Adding a VM](#adding-a-vm)
+   1. Make sure image is either Debian 9 or Ubuntu 16.04
+   2. Select your previously uploaded SSH public key in the form
+   3. Ssh and Web security groups should be assigned 
+   4. "Auto-assign a floating IP" option in Networks combobox should be selected (or any existing IP address)
+4. Wait for VM to start.
+
+>If VM status is active, it does not mean that a Linux distribution has been installed, you need to wait a few moments before you will be allowed to connect to a VM.
+
+VM configuration is depicted on the following screenshot:
+![Configuration for Python and JupyterHub management services](../images/vm-config-for-python-jupyter-management.png)
+
+Now, navigate to Resources -> Applications on the left. Then press Create application in the top right corner of the table and select Python management:
+
+![Available applications popup](../images/available-applications-list.png)
+
+Before saving the form, follow the instructions under “Virtual machine has cloud broker public key” checkbox:
+
+1. Find out External IP of your VM
+   ![VM External IP](../images/vm-external-ip.png)
+
+2. Connect with PuTTY (private key should correspond to the public key you uploaded to ETAIS!) to your VM
+   ![Connect via Putty](../images/connect-putty.png)
+
+3. Or use OpenSSH on Linux/MacOS: ssh TARGET_USER@VM_EXTERNAL_IP
+
+Once you are logged in (default system user depends on a distribution: either debian or ubuntu), copy cloud broker public key from the Python management form and paste it into /home/YOUR_TARGET_SYSTEM_USER/.ssh/authorized_keys
+
+![Pasting ETAIS public key into authorized_keys on your VM](../images/python-management-authorized-keys.png)
+
+Now you can proceed with Python management form and afterwards cotinue with [JupyterHub management](#jupyterhub-management-service).
+
+### Details screen elements
+
+![Python management details screen elements](../images/python-management-details.png)
+
+> Virtual environments are managed using virtualenv and virtualenvwrapper utilities.
+
+Python Management service is not notified when you install additional libraries manually.
+
+> If you install any libraries directly or create new virtual environments via SSH, click either Find missing libraries or Find installed virtual environments & libraries.
+
+1. Button to propagate changes to a virtual machine (install/delete libraries, create/delete virtual environments)
```

Comment:
This block of labels is not readable, and it looks like a broken sentence, because there's no introductory sentence. It's very confusing and it's not clear which screenshot does it refer to?